pandas/core/internals/blocks.py

```diff
@@ -1661,7 +1661,7 @@ def _unstack(self, unstacker, fill_value, new_placement):
 class NumpyBlock(libinternals.NumpyBlock, Block):
     values: np.ndarray
 
-    getitem_block_index = libinternals.NumpyBlock.getitem_block_index
+    getitem_block_index = final(libinternals.NumpyBlock.getitem_block_index)
```

Comment:
@jbrockmendel why is the override needed? is this a cython class thing.

Reply:
IIRC this was to make mypy happy

Reply:
the mypy errors are...

```
pandas/core/internals/blocks.py:1661: error: Definition of "getitem_block_index" in base class "NumpyBlock" is incompatible with definition in base class "Block"  [misc]
pandas/core/internals/blocks.py:1670: error: Definition of "getitem_block_index" in base class "NDArrayBackedBlock" is incompatible with definition in base class "Block"  [misc]
Found 2 errors in 1 file (checked 1299 source files)
```

they aren't needed. fixed up mypy error by moving remaining Python implementation of `getitem_block_index` from `Block` to `ExtensionBlock` and added `@final` to stub instead.

Also added `@final` to Python implementation of `getitem_block_index` for consistency.